File: python/penjualMobilBekas.py
from orang import orang
from mobil import mobil

class penjualMobilBekas(orang):
    def __init__(self, no_ktp="", nama="", alamat="", no_izin_usaha="", list_mobil=None):
        super().__init__(no_ktp, nama, alamat)
        self.no_izin_usaha = no_izin_usaha
        self.list_mobil = list_mobil if list_mobil is not None else []

    # Getter methods
    def get_no_izin_usaha(self):
        return self.no_izin_usaha

    def get_list_mobil(self):
        return self.list_mobil

    # Setter methods
    def set_no_izin_usaha(self, no_izin_usaha):
        self.no_izin_usaha = no_izin_usaha

    def set_list_mobil(self, list_mobil):
        self.list_mobil = list_mobil

File: cpp/penjualMobilBekas.cpp
#pragma once
#include <bits/stdc++.h>
#include <iostream>
#include <string>
#include "orang.cpp"
#include "kendaraan.cpp"

using namespace std;

class penjualMobilBekas : public orang
{
    private:
        string no_izin_usaha;
        vector <mobil*> listMobil;

    // CONSTRUCTOR
    public:
        penjualMobilBekas(){
        }

        penjualMobilBekas(string noKTP, string nama, string alamat, string no_izin_usaha, vector <mobil*> listMobil)
        : orang(noKTP, nama, alamat){
            this->no_izin_usaha = no_izin_usaha;
            this->listMobil = listMobil;
        }

        // DESTRUCTOR
        ~penjualMobilBekas(){
        }

    // GETTER
    string get_no_izin_usaha(){
        return no_izin_usaha; 
    }

    vector<mobil*> get_list_mobil(){
        return listMobil; 
    }


    // SETTER
    void set_no_izin_usaha(string no_izin_usaha){
        this->no_izin_usaha = no_izin_usaha;
    }

    void set_list_mobil(vector <mobil*> listMobil){
        this->listMobil = listMobil;
    }

};
